### DISCUSSION TITLE
How to install OpenSIS on Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by klmartinson17** [OS4ED/openSIS-Classic#264] [General]
Installation procedure:
1. Install LAMP: https://upcloud.com/community/tutorials/installing-lamp-stack-ubuntu/

2. Prevent a database connect error
sudo mysql -u root                           # I had to use "sudo" since it was a new installation
mysql> USE mysql;
mysql> UPDATE user SET plugin='mysql_native_password' WHERE User='root';
mysql> FLUSH PRIVILEGES;
mysql> exit;
sudo service mysql restart

3. Install Git and Curl, and clone opensis
cd /var/www/html
sudo apt install git
sudo git clone https://github.com/os4ed/opensis-classic
sudo mv opensis-classic opensis
sudo apt install curl

4. Find out what user is running apache:
ps aux | grep -E '[a]pache|[h]ttpd|[_]www|[w]ww-data|[n]ginx' | grep -v root | head -1
(the user is most likely www-data. It may also be: apache or httpd)

If you have made a user account: yourusername,  (replace the "yourusername" below with the non-root account you created when you installed Ubuntu)
sudo chown -R yourusername:www-data ./opensis
sudo chmod -R g+rw ./opensis

5.
sudo nano /etc/mysql/conf.d/strict_mode.cnf

Paste the following code into that file:
[mysqld]
sql_mode=IGNORE_SPACE,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
Save and close the file. 
And then restart MySQL server:
sudo service mysql restart

**Comment by pmetieh** [comment]
Thanks for the installation guide. It was perfect. No hitches.

**Comment by openSISAdmin** [comment]
Thank you Ken for providing this info. We will add it to the support portal.
